common/system/tasks/packet_encoder.cpp

```diff
@@ -0,0 +1,44 @@
+#include "packet_encoder.h"
+
+#include <errno.h>
+#include <string.h>
+
+int FSKPacketRadioEncoder::Encode(RadioDecodedPacket_s& input,
```

Comment:
Some/all of the `input` parameters in this file look like they might be able to be `const`. Can you add that if they can be?

Reply:
so FecDecode4 actually mutates in place to deinterleave each quartet of bytes. Should I make it deinterleave into a class member instead of mutating the input array?